crates/core/src/masp.rs

```diff
@@ -478,23 +488,82 @@ impl Display for TransferSource {
     }
 }
 
+/// Represents the pre-image to a TransparentAddress
+#[derive(Debug, Clone, BorshDeserialize, BorshSerialize, BorshDeserializer)]
+pub enum TAddrData {
+    /// A transparent address
+    Addr(Address),
+    /// An IBC address
+    Ibc(String),
+}
+
+impl TAddrData {
+    /// Get the transparent address that this target would effectively go to
+    pub fn effective_address(&self) -> Address {
+        match self {
+            Self::Addr(x) => x.clone(),
+            // An IBC signer address effectively means that assets are
+            // associated with the IBC internal address
+            Self::Ibc(_) => IBC,
+        }
+    }
+
+    /// Get the contained IBC receiver, if any
+    pub fn payment_address(&self) -> Option<String> {
```

Comment:
Name does not match comment? Should this be called `ibc_receiver_address` ?

Reply:
Yes, you're correct. Will fix this.